internal/services/contracts_events_consumer.go

```diff
@@ -145,3 +153,51 @@ func (c *ContractsEventsConsumer) setPrivilegeHandler(e *ContractEventData) erro
 
 	return nil
 }
+
+type MintedAftermarketDeviceArgs struct {
+	TokenID       string `mapstructure:"newTokenId"`
+	DeviceAddress string
+	Owner         string // TODO - confirm who the minter is
+}
+
+func (c *ContractsEventsConsumer) setMintedAfterMarketDevice(e *ContractEventData) error {
+	p := MintedAftermarketDeviceArgs{}
+	err := mapstructure.WeakDecode(e.Arguments, &p)
+	if err != nil {
+		return err
+	}
+
+	autopiApiService := NewAutoPiAPIService(c.settings, c.db.DBS)
+	device, err := autopiApiService.GetDeviceByEthAddress(p.DeviceAddress)
+	if err != nil {
+		c.log.Error().Msg(fmt.Sprintf("Couldn't fetch dongle with eth_address %s.", p.DeviceAddress))
+		return fmt.Errorf("couldn't fetch dongle with eth_address %s", p.DeviceAddress)
+	}
+
+	var maybeAddr null.Bytes
+
+	if strAddr := device.EthereumAddress; common.IsHexAddress(p.DeviceAddress) {
```

Comment:
If it's not an address then I would expect the call on 171 to fail; I doin't think you can mint without an address at this point. Maybe we check earlier than this?